### PR TITLE
create hot fix branch 2.7.9

### PR DESCRIPTION
**Pull request by rohitnarula7176**
*Hotfix branch 2.7.9

### CODE REVIEW

File: ADAL/src/workplacejoin/mac/ADWorkPlaceJoinUtil.m [hunk 1]
```diff
@@ -121,160 +118,85 @@ + (ADRegistrationInformation *)getRegistrationInformation:(id<MSIDRequestContext
...
-    if (*identity)
+    for (int resultIndex = 0; resultIndex < identityCount; resultIndex++)
     {
-        CFRelease(*identity);
+        identityDict = (NSDictionary *)CFArrayGetValueAtIndex(identityList, resultIndex);
+        
+        if ([identityDict isKindOfClass:[NSDictionary class]])
+        {
+            currentIssuer = [identityDict objectForKey:(__bridge NSString*)kSecAttrIssuer];
+            currentIssuerName = [[NSString alloc] initWithData:currentIssuer encoding:NSASCIIStringEncoding];
```
Comment (oldalton) [inline ADAL/src/workplacejoin/mac/ADWorkPlaceJoinUtil.m]: nit: can we return NSString issuer directly, and avoid extra conversion in the above code?

File: ADAL/src/workplacejoin/mac/ADWorkPlaceJoinUtil.m [hunk 2]
```diff
@@ -121,160 +118,85 @@ + (ADRegistrationInformation *)getRegistrationInformation:(id<MSIDRequestContext
...
+            
+            if ([challengeIssuerName caseInsensitiveCompare:currentIssuerName] == NSOrderedSame)
```
Comment (oldalton) [inline ADAL/src/workplacejoin/mac/ADWorkPlaceJoinUtil.m]: nit: please add a comment why it's a case insensitive compare for future reference

File: ADAL/src/workplacejoin/mac/ADWorkPlaceJoinUtil.m [hunk 3]
```diff
@@ -51,58 +50,56 @@ + (ADRegistrationInformation *)getRegistrationInformation:(id<MSIDRequestContext
...
         {
+            // The code above will catch missing security items, but not missing item attributes. These are caught here.
+            ADAuthenticationError* adError =
+            [ADAuthenticationError unexpectedInternalError:@"Missing some piece of WPJ data"
+                                             correlationId:context.correlationId];
+            
             *error = adError;
         }
+        
         goto _error;
```
Comment (oldalton) [inline ADAL/src/workplacejoin/mac/ADWorkPlaceJoinUtil.m]: the question I had is where else this goto is being used other than this one place?

Comment (rohitnarula7176) [reply ADAL/src/workplacejoin/mac/ADWorkPlaceJoinUtil.m]: This goto is used in the macro as discussed offline.

File: ADAL/src/workplacejoin/mac/ADWorkPlaceJoinUtil.m (hunk 2 above)
Comment (rohitnarula7176) [reply ADAL/src/workplacejoin/mac/ADWorkPlaceJoinUtil.m]: Sure. Addressed.

File: ADAL/src/workplacejoin/mac/ADWorkPlaceJoinUtil.m (hunk 1 above)
Comment (rohitnarula7176) [reply ADAL/src/workplacejoin/mac/ADWorkPlaceJoinUtil.m]: Good point. Addressed.

**Pull request merged**